msdk-rawdata/msdk-rawdata-peakinvestigator/src/main/java/io/github/msdk/rawdata/peakinvestigator/PeakInvestigatorMsSpectrum.java

```diff
@@ -0,0 +1,125 @@
+/* 
+ * (C) Copyright 2015-2016 by MSDK Development Team
+ *
+ * This software is dual-licensed under either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+
+package io.github.msdk.rawdata.peakinvestigator;
+
+import java.util.TreeMap;
+
+import com.google.common.collect.Range;
+
+import io.github.msdk.datamodel.msspectra.MsSpectrum;
+import io.github.msdk.datamodel.msspectra.MsSpectrumType;
+import io.github.msdk.util.tolerances.MzTolerance;
+
+/**
+ * This class is used provide m/z-dependent mass tolerances from
+ * PeakInvestigator by decorating an existing instance of MsSpectrum.
+ *
+ */
+public class PeakInvestigatorMsSpectrum implements MsSpectrum, MzTolerance {
```

Comment:
Also a good point.

I was hoping to map the PI workflow to standard centroiding as closely as possible (i.e. the resulting ```RawDataFile``` can be transparently passed to other Methods like ```ChromatogramBuilder```), but still have the mass errors around if desired. So while a custom ```MsSpectrum``` probably isn't necessary, I think a custom ```RawDataFile``` that either implements ```MzToleranceProvider``` or has a getter for one will be necessary.

What do you think?